Model/Cron.php

```diff
@@ -382,26 +382,68 @@ public function processNotification()
-        $minutesUntilProcessing = floor(($createdAt->getTimestamp() - $offerClosedMinDate->getTimestamp()) / 60);
-        if ($notification['event_code'] == Notification::OFFER_CLOSED && $minutesUntilProcessing > 0) {
-            $this->_adyenLogger->addAdyenNotificationCronjob(
-                sprintf(
-                    'OFFER_CLOSED notification %s skipped! Wait %s minute(s) before processing.',
-                    $notification->getEntityId(),
-                    $minutesUntilProcessing
-                )
-            );
+        switch ($notification['event_code']) {
+            // Remove OFFER_CLOSED and AUTHORISATION success=false notifications for some time from the processing list
+            // to ensure they won't close any order which has an AUTHORISED notification arrived a bit later than the
+            // OFFER_CLOSED or the AUTHORISATION success=false one.
+            case Notification::OFFER_CLOSED:
+                // OFFER_CLOSED notifications needs to be at least 10 minutes old to be processed
+                $offerClosedMinDate = new \DateTime();
+                $offerClosedMinDate->modify('-10 minutes');
+                $createdAt = \DateTime::createFromFormat('Y-m-d H:i:s', $notification['created_at']);
+                // To get the difference between $offerClosedMinDate and $createdAt, $offerClosedMinDate time in seconds
+                // is deducted from $createdAt time in seconds, divided by 60 and rounded down to integer
+                $minutesUntilProcessing = floor(
+                    ($createdAt->getTimestamp() - $offerClosedMinDate->getTimestamp()) / 60
+                );
 
-            return true;
+                if ($minutesUntilProcessing > 0) {
+                    $this->_adyenLogger->addAdyenNotificationCronjob(
+                        sprintf(
+                            'OFFER_CLOSED notification (entity_id: %s) for merchant_reference: %s is skipped! Wait %s minute(s) before processing.',
+                            $notification->getEntityId(),
+                            $notification->getMerchantReference(),
+                            $minutesUntilProcessing
+                        )
+                    );
+
+                    return true;
+                }
+
+                break;
+            case Notification::AUTHORISATION:
+                // Only delay success=false notifications processing
+                if (
+                    strcmp($notification['success'], 'true') == 0 ||
+                    strcmp($notification['success'], '1') == 0
+                ) {
+                    // do not skip this notification but process it now
+                    return false;
+                }
+
+                // AUTHORISATION success=false notifications needs to be at least 10 minutes old to be processed
+                $authorisationSuccessFalseMinDate = new \DateTime();
```

Comment:
Maybe we should combine both switch cases or move the timestamp generation/comparison outside the switch as the logic is pretty similar? I think the only difference is the log message

Reply:
was thinking to move the `$createdAt = \DateTime::createFromFormat('Y-m-d H:i:s', $notification['created_at']);` line outside of the switch but the -10 min calculation I wanted to keep notification specific so if needed we can change the minutes per notification type.

Reply:
I see, sounds good. Maybe when the cron is refactored we have notification classes and with a delayTime property

Reply:
yes definitely the refactored version of this fix is already defined here: PW-3638